libcontainerd/container_linux.go

```diff
@@ -191,3 +192,16 @@ func (ctr *container) handleEvent(e *containerd.Event) error {
 	}
 	return nil
 }
+
+func (ctr *container) discardFifos() {
+	for _, i := range []int{syscall.Stdout, syscall.Stderr} {
+		f := ctr.fifo(i)
+		c := make(chan struct{})
+		go func() {
+			close(c) // this channel is used to not close the writer too early, before readonly open has been called.
```

Comment:
Is the race between closing this and actually calling `io.Copy` (or is it `openReaderFromFile`?) ok?